### ISSUE TITLE
MNIST based Liner model - run inference e2e

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-mlir#149]
### Summary
Before running training, we should complete inference e2e for the based Linear model. 

Few dependencies before pushing this effort:
- #77
  - Softmax mustn't be decomposed - utilize ttnn softmax
- #145 
  - Most of the ops are working e2e. Important ops are the ones from the fwd graph (add, matmul, softmax)

### Details
By completing dependencies, most of the requirements should be in place. As a next step, feel free to check this test `pybuda/test/mlir/mnist/test_inference.py`. It needs to be modified to use `pybuda.compile` instead of `torch.compile` + maybe a few PyBuda-related cleanups (transposes, nops, etc). But mostly, with low effort it should work out of the box. 

### Passing criteria
- Whole models runs e2e
- Golden/Silicon PCC matches
- Model outputs valid results
  - #150 as requirement

**Comment by nvukobratTT** [comment]
Separated milestone is created for  e2e inference: https://github.com/orgs/tenstorrent/projects/54/views/6?sliceBy%5Bvalue%5D=%5BTT-Forge%5D+-+MNIST+E2E+Inference
